### ISSUE TITLE
R update to 3.2.3

### ISSUE THREAD

**Issue by verdurin** [hpcugent/easybuild-easyconfigs#2137]
I'm working on updating the Bioconductor bundle to 3.2, and at least some of the R packages require R 3.2.2.

Could we make that update? I'm happy to do that myself, if others agree.

**Comment by boegel** [comment]
@verdurin: I see no reason why you can't look into that. :-)
Ideally, all the R packages included in the R easyconfig should also be updated to their latest version.

There is some scripting available for that somewhere, don't do that by hands unless you're OK with going nuts on this. :P

**Comment by verdurin** [comment]
Having been doing it manually so far, I would very much welcome a pointer to any relevant scripts...

**Comment by sjtbham** [comment]
I had a go at doing a 3.2.3 package (yes I manually went down and sorted the versions of extensions as well as re-ordering and new deps). I've managed to build it with the boss 2015b toolchain, however I can't get it to build with iomkl 2015.03 (I don't have intel MPI, so can't try a straight intel 2015b one).

When building with iomkl, I get:
installing to /gpfs/apps/EL7-haswell/software/R/3.2.3-iomkl-2015.03/lib64/R/library/Rcpp/libs
** R
** inst
** preparing package for lazy loading
code for methods in class “C++OverloadedMethods” was not checked for suspicious field assignments (recommended package ‘codetools’ not available?)
code for methods in class “RcppClass” was not checked for suspicious field assignments (recommended package ‘codetools’ not available?)
code for methods in class “RcppClass” was not checked for suspicious field assignments (recommended package ‘codetools’ not available?)
** help
*** installing help indices
** building package indices
** installing vignettes
** testing if installed package can be loaded
Error in dyn.load(file, DLLpath = DLLpath, ...) : 
  unable to load shared object '/gpfs/apps/EL7-haswell/software/R/3.2.3-iomkl-2015.03/lib64/R/library/Rcpp/libs/Rcpp.so':
  /lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /gpfs/apps/EL7-haswell/software/R/3.2.3-iomkl-2015.03/lib64/R/library/Rcpp/libs/Rcpp.so)
Error: loading failed
Execution halted
ERROR: loading failed
 (at easybuild/EL7/software/EasyBuild/2.5.0/lib/python2.7/site-packages/easybuild_framework-2.5.0-py2.7.egg/easybuild/tools/run.py:397 in parse_cmd_output)
== 2016-01-09 16:14:14,979 runpy.easyblock WARNING build failed (first 300 chars): cmd "R CMD INSTALL /gpfs/sysapps/easybuild/UoB/sources/r/R/extensions/Rcpp_0.12.2.tar.gz   --library=/gpfs/apps/EL7-haswell/software/R/3.2.3-iomkl-2015.03/lib64/R/library --no-clean-on-error" exited with exitcode 1 and output:
* installing *source* package ‘Rcpp’ ...
** package ‘Rcpp’ succes
== 2016-01-09 16:14:14,980 runpy.EB_R INFO Closing log for application name R version 3.2.3


I'm not sure why its finding the system libstc++ here rather than the one form the iomkl toolchain in the GNU compiler part of that ... anyone any suggestions?

I've attached the log from the build as well as the .easy configs for both foss and iomkl.

Note that I've added a few deps which I guess were coming from OS installed packages before? (My build machine is a fairly cut down install). The GDAL components are currently commented out as I can't get the required GDAL to build at the moment (some MPI linker problem).


[R-3.2.3-iomkl-2015.03.eb.txt](https://github.com/hpcugent/easybuild-easyconfigs/files/83742/R-3.2.3-iomkl-2015.03.eb.txt)
[R-3.2.3-foss-2015b.eb.txt](https://github.com/hpcugent/easybuild-easyconfigs/files/83743/R-3.2.3-foss-2015b.eb.txt)

[easybuild-R-3.2.3-20160109.155226.GYxrI.log.txt.gz](https://github.com/hpcugent/easybuild-easyconfigs/files/83744/easybuild-R-3.2.3-20160109.155226.GYxrI.log.txt.gz)

**Comment by boegel** [comment]
@sjtbham: great, thanks for going through the effort and sharing it!

Maybe trying to statically link libstdc++ in Rcpp can help, cfr. http://stackoverflow.com/a/25482925/1467322

**Comment by boegel** [comment]
@verdurin: I never answered your question... There was some talk on this on the ML, but it seems like the script that Pieter was working on never surfaced.

Maybe revive that thread?

**Comment by verdurin** [comment]
Just tried that easyconfig and I see the following dependency errors:

```
[centos@easybuild-6 tmp]$ eb ./R-3.2.3-foss-2015b.eb -dr
== temporary log file in case of crash /tmp/centos/eb-3sxOhy/easybuild-C_Gm0a.log
== resolving dependencies ...
ERROR: Irresolvable dependencies encountered: libpng/1.6.20-foss-2015b, libjpeg-turbo/1.4.2-foss-2015b, cURL/7.46.0-foss-2015b, libxml2/2.9.3-foss-2015b
```

**Comment by boegel** [comment]
@verdurin: can this be closed and followed up in #2364?

**Comment by verdurin** [comment]
Yes, makes sense.
